src/core.js

```diff
@@ -563,5 +564,12 @@ Sequence.prototype._finally = function Sequence$finally(other){
 Sequence.prototype._fork = interpreter;
 
 Sequence.prototype.toString = function Sequence$toString(){
-  return `${this._spawn.toString()}${this._actions.map(x => `.${x.toString()}`).join('')}`;
+  let str = this._spawn.toString(), tail = this._actions;
+
+  while(!tail.isEmpty){
+    str = `${str}.${tail.head.toString()}`;
```

Comment:
We can safely avoid unnecessary dispatching, because we *know* that every item in `this._actions` has a custom `toString` function defined which we want to use. If not, that would be a bug in Fluture.